pinot-spi/src/test/java/org/apache/pinot/spi/env/CommonsConfigurationUtilsTest.java

```diff
@@ -50,6 +53,33 @@ public void tearDown()
     FileUtils.deleteDirectory(TEMP_DIR);
   }
 
+  @Test
+  public void testSegmentMetadataFromFile() {
+    // load the existing config and check the properties config instance
+    try {
+      PropertiesConfiguration config = CommonsConfigurationUtils
+          .getSegmentMetadataFromFile(SEGMENT_METADATA_CONFIG_FILE, true);
+      assertNotNull(config);
+
+      config.setProperty("testKey", "testValue");
+
+      // add the segment version header to the file and read it again
+      CommonsConfigurationUtils.saveSegmentMetadataToFile(config, CONFIG_FILE,
+          CommonsConfigurationUtils.PROPERTIES_CONFIGURATION_HEADER_VERSION_2);
+
+      // reading the property with header.
+      config = CommonsConfigurationUtils.getSegmentMetadataFromFile(CONFIG_FILE, true);
+      assertNotNull(config);
+      assertEquals(config.getHeader(), "# version=2");
+    } catch (Exception ex) {
+      Assert.fail("should not throw ConfigurationException exception with valid file");
+    }
+
+    // load the non-existing file and expect the exception
+    Assert.expectThrows(NullPointerException.class,
+        () -> CommonsConfigurationUtils.getSegmentMetadataFromFile(null, true));
```

Comment:
can add another check for non-existing file, in addition to `null`

Reply:
we don't need the non-existing file test.

Reply:
> we don't need the non-existing file test.

is this because we do `fileHandler.setFile(file);` if file doesn't exist?

not introduced by this PR, but looks like you added those changes earlier on, so in case you know: 
1. for `fromFile()`, why we need to do `fileHandler.setFile(file)` when file doesn't exit?
2. if that's needed, for `fromPath()` method, should we do `fileHandler.setPath(path)` if path doesn't exist as well?

Reply:
addressed in the [PR](https://github.com/apache/pinot/pull/13201). marking it as resolved.